### PR TITLE
recommendation component first draft

### PR DESCRIPTION
**Pull request by jpmarceau**
PR to review the code I have so far for the SF tutorial. It's heavily inspired from Etienne's recommendations component in the CDK. The main differences are that I use the JSUI `class="CoveoRecommendation"` (rather than `class="CoveoSearchInterface"`), and that I fetch the user profile in the database to pass it as a context value. It seems to work but this is my first SF code PR so your reviews would be really helpful hehe

I haven't really touched the CSS in this PR, I'll do so in a later one. Here I'd mostly like to validate the skeleton/logic.

I'm not sure what are the best practices when getting SF code reviewed. If there'a a good way to share scratch org demos for example, let me know.

Here are links to the write up I have so far, if it could be useful: 
https://github.com/coveo/coveo-training-content/compare/C101-601
https://levelupdev.coveo.com/authoring/branches/C101-601/learn/courses/create-custom-recommendation-component

### CODE REVIEW

File: force-app/main/default/aura/customRecommendations/customRecommendations.css [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.THIS {
```
Comment (jpmarceau) [inline force-app/main/default/aura/customRecommendations/customRecommendations.css:1]: Haven't added styling yet, I plan to do so last in another PR.

File: force-app/main/default/aura/customRecommendations/customRecommendations.cmp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+   <div>
+       <article>
+           <header>
+                <h2>
+                    <span>
+                            {!v.title}
+                    </span>
+                </h2>
+           </header>
+            <CoveoV2:SearchUI name="customRecommendations"
```
Comment (lbergeron) [inline force-app/main/default/aura/customRecommendations/customRecommendations.cmp:29]: You should use `name="{!v.name}"` here and add `customRecommendations` as the default value for the name attribute.

File: force-app/main/default/classes/userProfileHelper.cls [hunk 3]
```diff
@@ -0,0 +1,7 @@
+public with sharing class userProfileHelper {
```
Comment (lbergeron) [inline force-app/main/default/classes/userProfileHelper.cls]: IMO this class should be named `UserProfileController` instead.

File: force-app/main/default/classes/userProfileHelper.cls [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+    @AuraEnabled
+    public static Map<String,Object> getUserProfile() {
+        List<Profile> myProfile = [SELECT Id, Name FROM Profile WHERE Id=:userinfo.getProfileId() LIMIT 1];
```
Comment (lbergeron) [inline force-app/main/default/classes/userProfileHelper.cls]: You should perform [field security checks](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_perms_enforcing.htm) on `Profile.Id` and `Profile.Name` before attempting to execute the SOQL query.

File: force-app/main/default/aura/customRecommendations/customRecommendations.cmp (hunk 2 above)
Comment (jpmarceau) [reply force-app/main/default/aura/customRecommendations/customRecommendations.cmp:29]: Oups thank you
Actually I think I should have removed the attribute instead, I don't really plan on making many visualforce pages available to the user. `customRecommendations` will be the only one.
I'll fix in next commit

File: force-app/main/default/classes/userProfileHelper.cls (hunk 4 above)
Comment (jpmarceau) [reply force-app/main/default/classes/userProfileHelper.cls]: Thanks, I had no idea!
